app/tests/integration/nimbus/pages/base.py

```diff
@@ -12,13 +13,25 @@ class Base(Page):
     _save_continue_btn_locator = (By.CSS_SELECTOR, "#save-and-continue-button")
 
     def __init__(self, selenium, base_url, **kwargs):
-        super(Base, self).__init__(selenium, base_url, timeout=30, **kwargs)
+        super(Base, self).__init__(selenium, base_url, timeout=60, **kwargs)
 
     def wait_for_page_to_load(self):
         self.wait.until(EC.presence_of_element_located(self._page_wait_locator))
-
         return self
 
+    def wait_with_refresh(self, locator, message):
+        def _wait_for_it(selenium):
+            try:
+                self.wait_for_page_to_load()
+                selenium.find_element(*locator)
+            except NoSuchElementException:
+                selenium.refresh()
+                return False
+            else:
+                return True
+
+        self.wait.until(_wait_for_it, message=message)
+
```

Comment:
🔥

Reply:
No the built in wait is very nice and has fixed timeouts, the duration is specified [here](https://github.com/mozilla/experimenter/pull/6339/files#diff-2bb50b01c7c0f75eb7f6408f89b9c14ae62d0d5ecacdb072aacd8f8014595ae1R16) and it'll display the handy message when it fails :D